src/components/MainPage/MainPage.js

```diff
@@ -84,7 +98,7 @@ class MainPage extends Component {
             <SearchBar />
           </div>
           <div className="col-12 col-md-6 col-lg-4">
-            <Map afterClick={this.setCoords} />
+            <Map afterClick={this.setCoords} searchMethod={this.setSearchMethod} getNameRegion={this.getNameRegion} />
```

Comment:
afterClick, searchMethod and getNameRegion should be only one. It is more logic for the Map to only have a single click callback as all your callbacks are in the click event. The parent can then pass a single function with all the logic it needs.

```js
handleMapClick(coords, region) {
    this.setState({
      searchMethod: "map",
      region,
      coords,
    });
}

<Map afterClick={this.handleMapClick} />
```